include/irr/core/math/floatutil.h

```diff
@@ -48,39 +48,55 @@ IRR_FORCE_INLINE matrix4SIMD ROUNDING_ERROR<matrix4SIMD>();
 #endif
 //! Constant for PI.
 template<typename T>
-IRR_FORCE_INLINE T PI()
+IRR_FORCE_INLINE constexpr T PI()
 {
 	return T(3.14159265358979323846);
 }
 
 //! Constant for reciprocal of PI.
 template<typename T>
-IRR_FORCE_INLINE T RECIPROCAL_PI()
+IRR_FORCE_INLINE constexpr T RECIPROCAL_PI()
 {
 	return T(1.0) / PI<T>();
 }
 
 //! Constant for half of PI.
 template<typename T>
-IRR_FORCE_INLINE T HALF_PI()
+IRR_FORCE_INLINE constexpr T HALF_PI()
 {
 	return PI<T>() * T(0.5);
 }
 
+namespace impl
+{
+    constexpr uint32_t NAN_U32      = 0x7F800000u;
+    constexpr uint32_t INFINITY_U32 = 0x7FFFFFFFu;
```

Comment:
this is the wrong way around, NAN has all bits set, INF has 0s in the whole mantissa.

Reply:
@Crisspl @AnastaZIuk

Reply:
fixed in @Crisspl 's latest OBJ branch